src/test/java/org/jpeek/metrics/Lcom4Test.java

```diff
@@ -0,0 +1,165 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jpeek.metrics;
+
+import org.cactoos.list.ListOf;
+import org.junit.Ignore;
+import org.junit.Test;
+
+/**
+ * Test case for LCOM4
+ * LCOM4 = Logical Relatedness of Methods.
+ * Basically it's a graph disjoint set.
+ * I.e. it answers the the following question:
+ * "Into how many independent classes can you split this class?"
+ * @author Ilya Kharlamov (ilya.kharlamov@gmail.com)
+ * @version $Id$
+ * @since 0.28
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+public final class Lcom4Test {
+
+    /**
+     * Variable from the report.
+     */
+    public static final String PAIRS = "pairs";
+
+    /**
+     * Variable from the report.
+     */
+    public static final String ATTRIBUTES = "attributes";
+
+    /**
+     * Variable from the report.
+     */
+    public static final String METHODS = "methods";
+
+    /**
+     * Path to XSL.
+     */
+    private static final String XSL = "org/jpeek/metrics/LCOM4.xsl";
+
+    /**
+     * Tests the deep dependencies methodFour() -> methodTwo() -> methodOne().
+     *  MethodMethodCalls.java
+     *  - methodOne uses 'num' directly
+     *  - methodTwo uses 'num' indirectly via methodOne
+     *  - methodThree uses 'num' directly
+     *  - methodFour uses 'num' indirectly via methodTwo and methodOne
+     *  - methodFive does not use 'num' (this is an orphan method, ignored)
+     *  Therefore the number of disjoint sets (LCOM4) should be 1
+     *  since all the methods that use num
+     *  @todo #215:30min LCOM4: Graph algorithm to determine disjoint sets.
+     *  Currently we can only identify the dependencies via only one graph hop
+     *  so we can't trace methodFour that uses 'num' indirectly
+     *  via methodTwo and methodOne.
+     *  This one will be tricky to implement,
+     *  Since graph algorithms in XSLT - that will be a challenge.
+     */
+    @Test
+    @Ignore
```

Comment:
@ilyakharlamov Why this test is ignored? If it waits for something to be implemented, you need to put this in a puzzle

Reply:
@paulodamaso I have. Lcom4Test.java[75] `@todo #215:30min LCOM4: Graph algorithm to determine disjoint sets.`. Do I need to do anything else in addition to `@todo #215:30min` ?

Reply:
@ilyakharlamov Yes, please state in the puzzle that whoever resolves it must remove the ignore tag from this test

Reply:
@paulodamaso Added in 9198518.  Calculating Disjoint sets (also known as Connected Components), requires an algorithm like depth-first search. Not sure if it's possible to implement DFS in XSLT at all. Will be possible to do after #296